ui/src/experiment/ExperimentsRouter.js

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import {
+  EuiPage,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiText,
+} from "@elastic/eui";
+import { PageTitle } from "../components/page/PageTitle";
+
+import useDynamicScript from "../hooks/useDynamicScript";
+import loadComponent from "../utils/remoteComponent";
+import { defaultExperimentEngine } from "../config";
+
+export const ExperimentsRouter = ({ projectId }) => {
+  // Retrieve script from host dynamically
+  const { ready, failed } = useDynamicScript({
+    url: defaultExperimentEngine.url,
+  });
+
+  if (!ready || failed) {
+    return (
+      <EuiPage>
+        <EuiPageBody>
+          <EuiPageHeader>
+            <EuiPageHeaderSection>
+              <PageTitle title="Experiments" />
+            </EuiPageHeaderSection>
+          </EuiPageHeader>
+          <EuiPageContent>
+            <EuiText size="s" color="subdued">
+              {!defaultExperimentEngine.url
+                ? "No default Experiment Engine configured"
+                : !ready
+                ? "Loading Experiment Exgine ..."
+                : "Failed to load Experiment Exgine"}
+            </EuiText>
+          </EuiPageContent>
+        </EuiPageBody>
+      </EuiPage>
+    );
+  }
+
+  // Load component from remote host
+  const Component = React.lazy(
+    loadComponent(defaultExperimentEngine.appName, "./ExperimentsLandingPage")
+  );
+
+  return (
+    <React.Suspense fallback="Loading Experiments">
+      <Component projectId={projectId} />
+    </React.Suspense>
+  );
```

Comment:
```suggestion
  // Load component from remote host
  const ExperimentsLandingPage = React.lazy(
    loadComponent(defaultExperimentEngine.appName, "./ExperimentsLandingPage")
  );
  
  return ready 
    ? (
      <React.Suspense fallback="Loading Experiments">
        <ExperimentsLandingPage projectId={projectId} />
      </React.Suspense>
    ) : (
      <EuiPage>
        <EuiPageBody>
          <EuiPageHeader>
            <EuiPageHeaderSection>
              <PageTitle title="Experiments" />
            </EuiPageHeaderSection>
          </EuiPageHeader>
          <EuiPageContent>
            <EuiText size="s" color="subdued">
              { failed 
                ? (!!defaultExperimentEngine.url 
                     ? "Failed to load Experiment Exgine" 
                     : "No default Experiment Engine configured"
                ) : "Loading Experiment Exgine ..."
              }              
            </EuiText>
          </EuiPageContent>
        </EuiPageBody>
      </EuiPage>
    );
```

Reply:
But more specifically, I wonder if this component should be broke down into two:
1) When `defaultExperimentEngine.url` is not configured at all then we don't need to try to load something from remote host and just render `"No default Experiment Engine configured"` msg
2) When `defaultExperimentEngine.url` is configured, then `useDynamicScript` and `loadComponent` are called and proper remote component is rendered.

Reply:
Thanks for the suggestion. Refactored this file.